opnode/test/system_test.go

```diff
@@ -34,6 +34,64 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func waitForTransaction(hash common.Hash, client *ethclient.Client, pollInterval, timeout time.Duration) (*types.Receipt, error) {
```

Comment:
Similar functionality exists in geth https://github.com/ethereum/go-ethereum/blob/111a1b73cfe87782ed34659f704bbea70b032e1e/accounts/abi/bind/util.go#L32

Reply:
With a `WithTimeout ctx` that's basically identical. The only thing that I wish it had was a faster poll interval (we run the tests at 1s L2 block times and 2s L1 block times)